pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -257,6 +260,33 @@ func (c *Reconciler) reconcileSKS(ctx context.Context, pa *pav1alpha1.PodAutosca
 	return sks, nil
 }
 
+func (c *Reconciler) metricService(pa *pav1alpha1.PodAutoscaler) (*corev1.Service, error) {
+	svcs, err := c.serviceLister.Services(pa.Namespace).List(labels.SelectorFromSet(map[string]string{
+		autoscaling.KPALabelKey:   pa.Name,
+		networking.ServiceTypeKey: string(networking.ServiceTypeMetrics),
+	}))
+	if err != nil {
+		return nil, err
+	}
+	var ret *corev1.Service
+	for _, s := range svcs {
+		// Found a match.
+		// TODO(vagababov): determine if this is better to be in the ownership check.
+		if s.Name == pa.Status.MetricsServiceName {
+			ret = s
+			continue
+		}
+		if metav1.IsControlledBy(s, pa) {
+			// If we don't control it, don't delete it.
```

Comment:
Can you reword this comment to match the sense of the predicate this is under?

e.g. `// If it's not the metrics service recorded in status, but we control it then it is a duplicate and should be deleted.`

Reply:
Done.